### PR TITLE
Add plugin: Embed Code File with Jupyter Notebook Support

### PR DESCRIPTION
**Pull request by mjg-foundation**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/tree/1.0.2

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

File: community-plugins.json [hunk 1]
```diff
@@ -12892,5 +12892,12 @@
         "author": "Technerium",
         "description": "Graph of the Number of Files in the Vault.",
         "repo": "technerium/obsidian-vault-size-history"
+    },
+    {
+        "id": "embed-code-file-ipynb-as-python",
+        "name": "Embed Code File with Jupyter Notebook Support",
+        "author": "Matt at Foundation",
+        "description": "This is a plugin for Obsidian that allows for embedding code files.",
```
Comment (mjg-foundation) [inline community-plugins.json:12900]:
```suggestion
        "description": "This is a plugin that allows for embedding code files.",
```

File: community-plugins.json [hunk 2]
```diff
@@ -12892,5 +12892,12 @@
         "author": "Technerium",
         "description": "Graph of the Number of Files in the Vault.",
         "repo": "technerium/obsidian-vault-size-history"
+    },
+    {
+        "id": "embed-code-file-ipynb-as-python",
+        "name": "Embed Code File with Jupyter Notebook Support",
+        "author": "Matt at Foundation",
+        "description": "This is a plugin that allows for embedding code files.",
```
Comment (mjg-foundation) [inline community-plugins.json:12900]:
```suggestion
        "description": "Embed code files, including Jupyter Notebooks displayed as python.",
```

**Comment by mjg-foundation** [comment]
> Thank you for your submission, an automated scan of your plugin code's revealed the following issues:
> ### Required
> 
> [[1](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/84723c92ec1d0aa9d5e7d047b0022b6e114b0929/main.ts#L156)]:Obsidian's configuration directory isn't necessarily `.obsidian`, it can be configured by the user. You can access the configured value from `Vault#configDir`
> 
> [[1](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/84723c92ec1d0aa9d5e7d047b0022b6e114b0929/main.ts#L162)][[2](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/84723c92ec1d0aa9d5e7d047b0022b6e114b0929/main.ts#L163)]:You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.
> 
> Do NOT open a new PR for re-validation. Once you have pushed all of the required changes to your repo, the bot will update the labels on this PR within 6 hours. If you think some of the required changes are incorrect, please comment with `/skip` and the reason why you think the results are incorrect.

/skip
- This detected `.obsidian` being used as part of a CSS class name. It isn't used to reference the configuation directory.
- The styles assigned using JavaScript are only applied to elements specific to this plugin, and they are taken from the plugin's settings, so this should not interfere with themes.

File: community-plugins.json [hunk 3]
```diff
@@ -12921,11 +12921,18 @@
         "description": "Correct words with the first two letters in uppercase.",
         "repo": "Ummler/obsidian-auto-correct-capitals"
     },
-	{
+	  {
```
Comment (mjg-foundation) [inline community-plugins.json:12924]:
```suggestion
	{
```

**Comment by mjg-foundation** [comment]
@joethei sorry to tag, but the [pipeline](https://github.com/obsidianmd/obsidian-releases/actions/runs/10433331983) for ObsidianReviewBot to assign this to someone for review failed. This is an improvement on an existing plugin, whose maintainer is unresponsive judging by their github activity, so I don't want to modify things outside the scope of my improvement like how styles are assigned.

**Comment by joethei** [comment]
What sets your plugin appart from the original plugin?
Why can't this be a PR to said plugin?

[Copyright (c) [year] [fullname]](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/LICENSE#L3)
Fill this, and credit the original author.

["description": "This is a plugin for Obsidian that allows for embedding code files.",](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/manifest.json#L6)
Avoid including sentences like "This is a plugin for Obsidian that does ..." in your description, it should be self-evident that this is a plugin for Obsidian.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

[console.log(`registering renderer for ${l}`)](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/main.ts#L21)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.


[containerEl.createEl('h2', {text: 'Embed Code File Settings'});](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/settings.ts#L33)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[.setName('Included Languages')](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/settings.ts#L36), [.setName('Jupyter Notebook Cell Numbers')](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/settings.ts#L77)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[.setName('Display Jupyter Notebooks as python')](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/settings.ts#L67)
use the proper capitalization of "Python"

[import path from "path";](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/utils.ts#L1)
This import is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[const result = path.join(dir, subpath);](https://github.com/mjg-foundation/embed-code-file-ipynb-as-python/blob/f7da9a1c669a0c6d90c888ea5bc50ef0356fe3ba/utils.ts#L4)
Just join them with regular slashes `/`, Obsidian uses that on all platforms.
Or, just remove that function, since it's not used.

**Comment by mjg-foundation** [comment]
@joethei I have made a PR to the original, but the maintainer is no longer active: https://github.com/almariah/embed-code-file/pull/19

I'll look into making your suggested changes to my additions, but I originally intended to only make minimal changes to the original, which had been approved as a plugin and included the other issues.

**Pull request closed without merging**